code/src/pages/Company.js

```diff
@@ -0,0 +1,81 @@
+import React, { useEffect, useState } from 'react'
+import { useParams, Link } from 'react-router-dom'
+import { LoadingSpinner } from 'components/LoadingSpinner'
+import { NotFound } from 'components/NotFound'
+import { BackToList } from 'components/BackToList'
+import './company.css'
+
+export const Company = () => {
+
+  //Destructuring to get the company id key from the url params, called id beacuse that is what we named it in the path in App.js
+  const { companyId } = useParams()
+
+  //My API key
+  const apiKey = "f16bd1845da3bcbe9df17f656b96d33b"
+
+  //Loading state to show loading mean while API is fetched
+  const [loading, setLoading] = useState(true)
+  //Need a state to map the data about company
+  const [company, setCompany] = useState([])
+  //Need a state to map the movies from that company
+  const [companyMovies, setCompanyMovies] = useState([])
+  //To set error state if movie not found
+  const [error, setError] = useState(false)
+
+  //Fetching company details
+  useEffect(() => {
+    setLoading(true)
+    fetch(`https://api.themoviedb.org/3/company/${companyId}?api_key=${apiKey}`)
+      .then((res) => res.json()) //Get the json of company
+      .then((json) => {
+        if (json.status_code === 34) {
+          setError(true)
+        } else {
+          setCompany(json)
+        }
+        setLoading(false)
+      })
+  }, [companyId])
+
+  //Fetching movies by company
+  useEffect(() => {
+    fetch(`https://api.themoviedb.org/3/discover/movie?api_key=${apiKey}&language=en-US&sort_by=popularity.desc&include_adult=false&include_video=false&page=1&&with_companies=${companyId}`)
+      .then((res) => res.json()) //Get the json of company
+      .then((json) => {
+        setCompanyMovies(json.results)
+      })
+  }, [companyId], [companyMovies])
+
+  return (
+    <main>
+      {loading && <LoadingSpinner />}
+
+      {error && <NotFound />}
+
+      {!loading && !error &&
+        <section className="companyWrapper">
+          <BackToList />
+          <div className="companyDetails">
+            {company.logo_path === null ? <h1>{company.name}</h1> : <img className="logo" src={`https://image.tmdb.org/t/p/w300${company.logo_path}`} alt={company.name} />}

```

Comment:
This is an intersting article regarding `null`, maybe you find it interesting! https://medium.com/javascript-in-plain-english/how-to-check-for-null-in-javascript-dffab64d8ed5